streamer/src/nonblocking/quic.rs

```diff
@@ -1171,43 +1190,31 @@ async fn handle_connection(
     stats.total_connections.fetch_sub(1, Ordering::Relaxed);
 }
 
-// Return true if the server should drop the stream
-async fn handle_chunk(
-    maybe_chunk: Option<quinn::Chunk>,
-    packet_accum: &mut Option<PacketAccumulator>,
-    remote_addr: &SocketAddr,
+// Handle the chunks received from the stream. If the stream is done, send the
+// packet to the packet sender.
+//
+// Returns Ok(true) if the stream is complete, Ok(false) if the stream is still
+// active, Err(()) if the stream is invalid and the peer should be disconnected.
+async fn handle_chunks(
```

Comment:
This function does two things:
1. processes new chunks if any
2. puts accumulated chunks into the channel

It looks to me that at the place where we do `match handle_chunks(` we already know will it do (1) or (2). So maybe, to simplify the logic a bit further, it could be decomposed into 2 functions each doing only one of these actions?

Reply:
This is a good point! I have another PR coming where I remove the channel altogether, so I'll leave this as is for now since I'm going to remove the code anyway.